### PR TITLE
[Algorithm] SOTA discrete offline CQL 

### LINKED ISSUE #3097: [Feature Request] Discrete offline CQL
## Motivation

Both offline and online in continuous/discrete action space CQL are implemented in sota-implementations, except for **discrete offline CQL**.

## Solution

Add support for discrete offline CQL. This would complete the CQL sota-implementations in the repository and benefit users working with discrete action space offline RL benchmarks and datasets.

## Alternatives

Use existing open-source discrete offline CQL implementations from other repositories.
However, this alternative may not maintain the same code quality, integration, or consistency as a native sota-implementation within torchrl.

## Additional context

Adding discrete offline CQL will bring the repository in line with other SOTA RL codebases and help the community benchmark and reproduce results for discrete action spaces in offline RL.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by Ibinarriaga8**
## Description

This PR solves issue [#3097](https://github.com/pytorch/rl/issues/3097) by providing a state-of-the-art (SOTA) implementation for discrete offline Conservative Q-Learning (CQL) within the repository.

- Implemented a SOTA discrete offline CQL algorithm fully integrated into the codebase.
- As with the SOTA implementation of discrete online CQL (already present in the repo), the CartPole environment was used for offline discrete CQL training.
- Leveraged the changes from [commit 0627e85c78756b4e4dbde726a9d5e85300e239a0](https://github.com/pytorch/rl/commit/0627e85c78756b4e4dbde726a9d5e85300e239a0) to utilize the updated `MinariExperienceReplay` class for loading experiences from a custom CartPole dataset into the offline replay buffer.
- Updated the discrete loss function according to the latest torchrl documentation, explicitly handling the categorical action space as required for discrete environments.
- Added a SOTA-check test for this new algorithm to ensure correctness and performance.

## Motivation and Context

This change was required to complete the set of SOTA CQL implementations in the repository, specifically by adding support for discrete offline CQL. Previously, only the online discrete CQL and both online/offline continuous CQL were available. By implementing discrete offline CQL, the repository now covers all major CQL benchmarks, enabling users to run and benchmark offline RL in discrete action spaces natively within torchrl.

The change also brings the repository in line with other top RL codebases and helps the community reproducibly benchmark and compare discrete offline RL algorithms.

This PR closes #3097.

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: sota-check/run_discrete_cql.sh
Comment (Ibinarriaga8) [inline sota-check/run_discrete_cql.sh:1]: Added sota-check

File: sota-implementations/cql/utils.py [hunk 1]
```diff
@@ -195,6 +195,49 @@ def make_offline_replay_buffer(rb_cfg):
     return data
```
Comment (Ibinarriaga8) [inline sota-implementations/cql/utils.py:197]: Replay buffer from a custom minari dataset

File: sota-implementations/cql/utils.py [hunk 2]
```diff
@@ -354,9 +397,11 @@ def make_continuous_loss(loss_cfg, model, device: torch.device | None = None):
 
 
 def make_discrete_loss(loss_cfg, model, device: torch.device | None = None):
+
     loss_module = DiscreteCQLLoss(
         model,
         loss_function=loss_cfg.loss_function,
+        action_space="categorical",
         delay_value=True,
```
Comment (Ibinarriaga8) [inline sota-implementations/cql/utils.py]: Updated make_discrete_loss to follow torchrl documentation

**Comment by Ibinarriaga8** [comment]
Log runs:
W B Chart 28_7_2025, 13_00_12

https://api.wandb.ai/links/202206789-universidad-pontificia-comillas/ir36pc30

File: sota-implementations/cql/discrete_online_config.yaml [hunk 3]
```diff
@@ -30,7 +30,7 @@ logger:
   eval_steps: 200
   mode: online
   eval_iter: 1000
-  video: False
+  video: True
```
Comment (vmoens) [inline sota-implementations/cql/discrete_online_config.yaml]: Do we want this?

**Review by vmoens — APPROVED** [review]
I think we're good to go when the linter is fixed!

**Comment by Ibinarriaga8** [comment]
@vmoens I made some changes to make sure SOTA checks work:

I applied minor fixes to the make_discrete_loss function in utils.py, and added the necessary commands in test_sota.py, including the sota-check for my new algorithm.
Please, lmk if anything else is needed for the sota-check

**Comment by Ibinarriaga8** [comment]
@vmoens I noticed that the minari dependency is missing from the linux_sota environment. Do you think it’s sufficient to simply add it to linux-sota/scripts/environment.yml?

**Comment by Ibinarriaga8** [comment]
SOTA-checks were successful 
Lmk if there is anything else needed
